Form/PhpcrOdmTypeGuesser.php

```diff
@@ -49,10 +50,16 @@ class PhpcrOdmTypeGuesser implements FormTypeGuesserInterface
 
     private $cache = array();
 
+    /**
+     * Work with 2.3-2.7 and 3.0 at the same time. drop once we switch to symfony 3.0.
+     */
+    private $entryTypeOption = 'type';
+
     public function __construct(ManagerRegistry $registry, $typeGuess = array())
     {
         $this->registry = $registry;
         $this->typeGuess = $typeGuess;
+        $this->entryTypeOption = Kernel::VERSION_ID < 20800 ? 'type' : 'entry_type';
```

Comment:
don't base this of the kernel version, the Form and HttpKernel components can have other versions. To check whether we are using Form 2.8 or 3.*, do something like `class_exists('Symfony\Component\Form\Util\StringUtil') ? symfony 2.8+ : symfony <2.8`